src/libYARP_dev/src/PolyDriver.cpp

```diff
@@ -286,15 +287,14 @@ bool PolyDriver::coreOpen(yarp::os::Searchable& prop) {
                     return false;
                 }
             }
-            if (creator!=NULL) {
-                ConstString name = creator->getName();
-                ConstString wrapper = creator->getWrapper();
-                ConstString code = creator->getCode();
-                yInfo("created %s <%s>. See C++ class %s for documentation.",
-                      ((name==wrapper)?"wrapper":"device"),
-                      name.c_str(),
-                      code.c_str());
-            }
+
+            ConstString name = creator->getName();
+            ConstString wrapper = creator->getWrapper();
+            ConstString code = creator->getCode();
+            yInfo("created %s <%s>. See C++ class %s for documentation.",
+                  ((name==wrapper)?"wrapper":"device"),
+                  name.c_str(),
+                  code.c_str());
```

Comment:
I don't understand this change, but I think this is a critical point of the device infrastructure. @drdanz did you checked this part?

Reply:
My reason: if `driver` is null, return false. But `driver` could be not null only if `creator` is not null. Coverity was complaining about `creator` being dereferenced at https://github.com/robotology/yarp/blob/7037d0b0bff7959c1c2d918dbbf238c2cf0a5dac/src/libYARP_dev/src/PolyDriver.cpp#L262 in between 2 null checks, so I avoided second null check on `creator` and put a direct `return false` if `creator` was null at first check.

Reply:
Got it.